core/http-auth-aws-crt/src/main/java/software/amazon/awssdk/http/auth/aws/crt/internal/signer/DefaultAwsCrtV4aHttpSigner.java

```diff
@@ -0,0 +1,142 @@
+        if (expirationDuration.compareTo(PRESIGN_URL_MAX_EXPIRATION_DURATION) > 0) {
+            throw new IllegalArgumentException(
+                "Requests that are pre-signed by SigV4 algorithm are valid for at most 7" +
+                " days. The expiration duration set on the current request [" + expirationDuration + "]" +
+                " has exceeded this limit."
+            );
+        }
+        return expirationDuration;
+    }
+
+    private static SyncSignedRequest doSign(SyncSignRequest<? extends AwsCredentialsIdentity> request,
+                                            AwsSigningConfig signingConfig,
+                                            V4aPayloadSigner payloadSigner) {
+        if (CredentialUtils.isAnonymous(request.identity())) {
+            return SyncSignedRequest.builder()
+                                    .request(request.request())
+                                    .payload(request.payload().orElse(null))
+                                    .build();
+        }
+
+        SdkHttpRequest sanitizedRequest = sanitizeRequest(request.request());
+
+        HttpRequest crtRequest = toRequest(sanitizedRequest, request.payload().orElse(null));
+
+        V4aContext v4aContext = sign(sanitizedRequest, crtRequest, signingConfig);
+
+        ContentStreamProvider payload = payloadSigner.sign(request.payload().orElse(null), v4aContext);
+
+        return SyncSignedRequest.builder()
+                                .request(v4aContext.getSignedRequest())
+                                .payload(payload)
+                                .build();
+    }
+
+    private static V4aContext sign(SdkHttpRequest request, HttpRequest crtRequest, AwsSigningConfig signingConfig) {
+        AwsSigningResult signingResult = CompletableFutureUtils.joinLikeSync(AwsSigner.sign(crtRequest, signingConfig));
+        return new V4aContext(
+            toRequest(request, signingResult.getSignedRequest())
+        );
+    }
+
+    @Override
+    public SyncSignedRequest sign(SyncSignRequest<? extends AwsCredentialsIdentity> request) {
+        AwsSigningConfig signingConfig = signingConfig(request);
+        V4aPayloadSigner payloadSigner = V4aPayloadSigner.create();
+        return doSign(request, signingConfig, payloadSigner);
+    }
+
+    private AwsSigningConfig signingConfig(SignRequest<?, ? extends AwsCredentialsIdentity> request) {
+        String regionName = request.requireProperty(REGION_NAME);
+        String serviceSigningName = request.requireProperty(SERVICE_SIGNING_NAME);
+        Clock signingClock = request.requireProperty(SIGNING_CLOCK, Clock.systemUTC());
+        boolean doubleUrlEncode = request.requireProperty(DOUBLE_URL_ENCODE, true);
+        boolean normalizePath = request.requireProperty(NORMALIZE_PATH, true);
+        AuthLocation authLocation = request.requireProperty(AUTH_LOCATION, AuthLocation.HEADER);
+        Duration expirationDuration = validateExpirationDuration(
```

Comment:
We need the default value, which it is the way it is.

Reply:
Yeah, you're right. I've changed it. We cannot have a default value, because we can't determine if it's a presigned case if we DO have a default value.